src/unix/fs.c

```diff
@@ -150,8 +146,16 @@ static ssize_t uv__fs_fdatasync(uv_fs_t* req) {
 
 static ssize_t uv__fs_fsync(uv_fs_t* req) {
 #if defined(__APPLE__)
-  /* See the comment in uv__fs_fdatasync. */
-  return fcntl(req->file, F_FULLFSYNC);
+  /* Apple's fdatasync and fsync explicitly do NOT flush the drive write cache
+   * to the drive platters. This is in contrast to Linux's fdatasync and fsync
+   * which do, according to recent man pages. F_FULLFSYNC is Apple's equivalent
+   * for flushing buffered data to permanent storage. If F_FULLFSYNC is not
+   * supported by the file system we should fall back to fsync(). This is the
+   * same approach taken by sqlite.
+   */
+  int r = fcntl(req->file, F_FULLFSYNC);
+  if (r) r = fsync(req->file);
```

Comment:
Should we do this unconditionally or just when the error returned confirms it's not supported?

Reply:
style:
```c
if (r != 0)
  r = fsync(req->file);
```

Reply:
Both SQLite and MySQL do this unconditionally. I'm not sure if the error code is documented anywhere. I received -25 on my system but I'm not sure if it's always the case. I think we should always fall back to `fsync()` if you are okay with that. If `fsync()` then fails we return the error from `fsync()`.

Regarding style, is this okay:

```
if (r != 0) r = fsync(req->file);
```

Reply:
Just used your style! Done.